src/components/Button/index.tsx

```diff
@@ -306,23 +306,36 @@ export const ButtonApprove = ({
   tooltipMsg,
   tokenSymbol,
   approval,
-  onClick,
+  approveCallback,
   disabled,
   forceApprove = false,
 }: {
   tooltipMsg: string
   tokenSymbol: string | undefined
   approval: ApprovalState
-  onClick: () => void
+  approveCallback: () => Promise<void>
   disabled: boolean
   forceApprove?: boolean
 }) => {
+  const loading = useRef(false)
```

Comment:
why do we use ref here? why dont we use useState, useRef doesnt trigger re-render after set loading.current = false

Reply:
I just want to prevent spam clicking button, don't need any rerender